src/8a860_len_3f30.c

```diff
@@ -1,11 +1,97 @@
 #include "common.h"
 
+extern s32 D_8010D650;
+extern s32 D_8010D660;
+extern s32 D_8010D664;
+extern s8 D_8010D699;
+extern PopupMenu* D_8010D69C;
+extern s32 D_8010D640;
+extern s32 D_8010D65C;
+extern s32 D_8010D668;
+extern s32 D_8010D66C;
+extern s32 D_8010D670;
+extern s32 D_8010D674;
+extern s32 D_8010D678;
+extern s8 D_8010D69A;
+extern s32 D_8010D648;
+extern s32 D_8010D64C;
+extern s8 D_8010D654;
+extern u8 D_8010D655;
+extern u8 D_8010D656;
+extern s8 D_8010D68E;
+extern s8 D_8010D68F;
+extern s8 D_8010D691;
+extern s8 D_8010D692;
+extern s32 D_8010D694;
+extern s8 D_8010D698;
+void func_800E9894(void);
+s8 func_800E98D4(void);
+void open_status_menu_short(void);
+void func_800E98C4(void);
+
+typedef s32 PopupType;
+enum PopupTypes {
+    POPUP_TYPE_USE_ITEM,
+    POPUP_TYPE_SWITCH_PARTNER,
+    POPUP_TYPE_THROW_AWAY_ITEM,
+    POPUP_TYPE_TRADE_FOR_BADGE,
+    POPUP_TYPE_UPGRADE_PARTNER,
+    POPUP_TYPE_SELL_ITEM,
+    POPUP_TYPE_CHECK_ITEM,
+    POPUP_TYPE_CLAIM_ITEM,
+    POPUP_TYPE_READ_LETTER,
+    POPUP_TYPE_TAKE_FROM_CHEST,
+    POPUP_TYPE_READ_DIARY_PAGE,
+    POPUP_TYPE_READ_POSTCARD,
+    POPUP_TYPE_USE_KEY,
+};
+
 INCLUDE_ASM(s32, "8a860_len_3f30", hide_popup_menu);
 
-INCLUDE_ASM(s32, "8a860_len_3f30", destroy_popup_menu);
+void destroy_popup_menu(void) {
+    free_hud_element(D_8010D65C);
+    free_hud_element(D_8010D678);
+    free_hud_element(D_8010D66C);
+    free_hud_element(D_8010D670);
+    free_hud_element(D_8010D674);
+
+    if (D_8010D69C->popupType == POPUP_TYPE_TRADE_FOR_BADGE) {
+        free_hud_element(D_8010D660);
+        free_hud_element(D_8010D664);
+    }
+
+    if (D_8010D69C->popupType == POPUP_TYPE_UPGRADE_PARTNER) {
+        free_hud_element(D_8010D660);
+        free_hud_element(D_8010D664);
+    }
+
+    if (D_8010D69C->popupType == POPUP_TYPE_SWITCH_PARTNER || D_8010D69C->popupType == POPUP_TYPE_UPGRADE_PARTNER) {
+        free_hud_element(D_8010D668);
+    }
+
+    if (D_8010D69C->popupType == POPUP_TYPE_SELL_ITEM) {
+        free_hud_element(D_8010D660);
+    }
+
+    if ((D_8010D69C->popupType <= POPUP_TYPE_USE_ITEM ||
```

Comment:
This `<` is weird. Does this match better as a switch?

Reply:
can't get it to match with a switch unfortunately